Polish up provisioners.mdx

Linda-Ikechukwu opened the pull request:
This pull request suggests several wording edits to the provisioner doc page: 
- Removing future tense in favor of present tense
- Change passive voice to active voice
- Changing plural to singular in several instances
- Spelling out terms in clear on the first use before introducing an acronym
- Several other wording and styling edits

Linda-Ikechukwu committed:
Update provisioners.mdx

Suggesting several wording edits. 
- Removing future tense in favor of present tense
- Changing plural to singular in several instances
- Spelling out terms in clear on the first use before introducing an acronym
- Several other wording and styling edits

Linda-Ikechukwu committed:
Update provisioners.mdx

tashian commented on step-ca/provisioners.mdx:
```diff
@@ -60,45 +60,60 @@ Azure  | ✔️  | ✔️  | 𝗫 | 𝗫 | ✔️  | 𝗫 | 𝗫 | 𝗫 | 𝗫
+To illustrate how to interpret this table, let's take the `JWK` and `SSHPOP` provisioner. 
+
+A `JWK` provisioner _can_ sign, renew, and revoke X.509 certificates, as well as sign user and host SSH certificates. 
+However, it _cannot_ renew or rekey SSH certificates.
```
```suggestion
However, it _cannot_ renew SSH user certificates.<Reference id="f1" marker="1"/>
```

tashian commented on step-ca/provisioners.mdx:
```diff
@@ -60,45 +60,60 @@ Azure  | ✔️  | ✔️  | 𝗫 | 𝗫 | ✔️  | 𝗫 | 𝗫 | 𝗫 | 𝗫
 ## Provisioner Management
 
-Use the [`step ca provisioner`](../step-cli/reference/ca/provisioner) command group to add, remove, or modify provisioner configurations. Run these commands directly on your CA machine. They need to modify the `$(step path)/config/ca.json` configuration file.
+To add, remove, or modify provisioner configurations, 
```
I'd suggest changing this section so that it begins by explaining that provisioner configuration can reside either in `ca.json`, or in the database. It resides in `ca.json` by default. Database-stored provisioners are good if you have multiple load-balanced CA nodes. Refer them to the [Remote Provisioner Management](https://smallstep.com/docs/step-ca/provisioners/#remote-provisioner-management) section for more about the tradeoffs between these two options.

The `step ca provisioner` commands try to detect where the provisioner configuration is stored. It will modify the `ca.json` file directly, or it will update the database, depending on whether you have enabled [Remote Provisioner Management](https://smallstep.com/docs/step-ca/provisioners/#remote-provisioner-management).

Linda-Ikechukwu replied:
Novice question: I've seen the phrase 'the database' across our docs. But I never quite understood what database is being referred to. Where is 'this' database emanates from? The term 'the database' insinuates that there is a singular database. Is this the case?

Linda-Ikechukwu replied:
Also, just noticed that the provisioner management section doesn't exactly discuss the tradeoffs or differences between the default and remote provisioner management options.

Linda-Ikechukwu replied:
I made changes in this pull request https://github.com/smallstep/docs/pull/239

tashian commented on step-ca/provisioners.mdx:
```diff
@@ -241,30 +260,31 @@ An administrator is just a subject name and a provisioner name.
-**Tip: Single Sign-On can be used for Admin users.** 
-To enable SSO CA administration, first add an [OIDC provisioner](#oauthoidc-single-sign-on) linked to your identity provider.
-Then, add an Admin user associated with that provisioner, using your email address as the Admin's subject name.
+***Tip: Single Sign-On (SS0) can be used for Admin users.*** 
```
```suggestion
***Tip: Single Sign-On (SSO) can be used for Admin users.*** 
```

tashian commented on step-ca/provisioners.mdx:
```diff
@@ -1014,19 +1083,24 @@ To get a client certificate for a hardware-bound private key on your YubiKey:
+The SCEP provisioner can sign and renew certificates 
+using the Simple Certificate Enrollment Protocol SCEP protocol (SCEP) ([RFC8894](https://datatracker.ietf.org/doc/html/rfc8894)). 
+SCEP is popularly used for certificate management, 
+particularly in network equipment and mobile device management (MDM). 
```
```suggestion
SCEP is commonly used for certificate management 
in network equipment and mobile device management (MDM). 
```

tashian commented:
Looking good @Linda-Ikechukwu, just added a few suggestions.

Linda-Ikechukwu committed:
Update step-ca/provisioners.mdx

Co-authored-by: Carl Tashian <carl@smallstep.com>

Linda-Ikechukwu committed:
Update step-ca/provisioners.mdx

Co-authored-by: Carl Tashian <carl@smallstep.com>

Linda-Ikechukwu committed:
Update step-ca/provisioners.mdx

Co-authored-by: Carl Tashian <carl@smallstep.com>

Linda-Ikechukwu committed:
Update provisioners.mdx

Linda-Ikechukwu committed:
Merge pull request #239 from smallstep/Linda-Ikechukwu-patch-2

Update provisioners.mdx

Linda-Ikechukwu committed:
Update provisioners.mdx

- Add the link to database page
- Add explanation for reason to delete JWK encrypted private key.

tashian reviewed (approved):
lgtm

tashian commented:
@Linda-Ikechukwu just saw this old PR here with your updates in it. Feel free to merge this one if you think it's ready
